cocotb/utils.py

```diff
@@ -107,16 +117,17 @@ def get_sim_steps(time, units=None):
     """
     result = time
     if units is not None:
-        result = result * (10.0**(_get_log_time_scale(units) - _LOG_SIM_PRECISION))
+        result = _ldexp10(result, _get_log_time_scale(units) - _LOG_SIM_PRECISION)
 
-    err = int(result) - math.ceil(result)
+    result_rounded = math.floor(result)
```

Comment:
Previously this would allow all negative numbers, but require exact matches for positive ones. That didn't look deliberate.